themes/default/content/blog/zephyr-code-placement-and-stacks/index.md

```diff
@@ -0,0 +1,133 @@
+
+As part of the switch to their new architecture, Zephyr has decided they want to use Pulumi to manage the infrastructure and application. As they prepared to embark on using Pulumi, a couple of questions came up for the Zephyr team:
+
+* Where should they store their Pulumi code? The ability for Zephyr's developers to use a full programming language to manage the infrastructure and deployment of their online store is a huge plus, but should this code go in the same repository as their application code? Or a different repository?
+* How do they address the need for multiple environments? It's pretty clear to the Zephyr development team that they'll need multiple instances of the online store (for things like testing, QA, or development). What's the best way to handle this when using Pulumi?
+
+Let's examine each of these issues in a bit more detail, and see how Zephyr is choosing to proceed.
+
+### Storing their Pulumi code
+
+When it comes to answering the question of where to store Pulumi code relative to the application(s) it supports, there are three basic options:
+
+1. In the same repository and same branch as your application code
+2. In the same repository as your application code, but in a different branch
+3. In a separate repository from your application code
+
+Each of these options has its own advantages and disadvantages. For example, ...
+
+Additionally, there are other considerations that might affect which decision an organization might choose.
+
+* Smaller organizations will typically have smaller teams that deal with both infrastructure as code artifacts as well as application development. In cases like this, using the same repository and same branch as the application code works well.
+* As organizations grow, however, teams may split and the responsibility for different parts of the code may diverge. In that case, it may make more sense for the Pulumi code to be stored separately from the application code so that appropriate access controls can be enforced. This is also typically the situation for large organizations where there are already separate teams for application development and platform engineering-type functions.
+
+{{% notes type="info" %}}
+The discussion above is working from the assumption that Pulumi users will store their Pulumi code in a version control system, like Git. Strictly speaking, using a version control system isn't required for Pulumi to function (there is one exception we'll touch on later in this series when we discuss Pulumi Deployments), but we **very strongly** recommend that all users use Pulumi in conjunction with version control.
+{{% /notes %}}
+
+Zephyr is a small organization with a single team of developers that are responsible for both managing the cloud infrastructure as well as the development of the online store application. For their use case, Zephyr felt like a single repository/same branch approach was most appropriate for their specific requirements.
```

Comment:
```suggestion
Zephyr felt like a single repository/same branch approach was most appropriate for their specific requirements: Zephyr is a small organization with a single team of developers that are responsible for both managing the cloud infrastructure as well as the development of the online store application.
```

(Changed the paragraph so that it starts with the choice and explains the reason why.)

Reply:
Incorporated in 35635a95